students/Mintae_K/lesson03/mailroom.py

```diff
@@ -0,0 +1,60 @@
+# Sending a Thank you
+def sendingthank(donors):
+    f_name = input('Please type the Full Name or list> ')
+    while f_name == 'list':
+        print(' ')
+        for x in donors:
+            print(x[0])
+        print(' ')
+        f_name = input('Please type the Full Name or list> ')
+    for x in range(len(donors)):
+        if f_name == donors[x][0]:
+            name_exists = True
+            break
+        else:
+            name_exists = False
+    if name_exists is False:
+        s = 'New donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors = donors + [[f_name, float(amount)]]
+    else:
+        s = 'Existing donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors[x] = donors[x] + [float(amount)]
+    print('\n', "Thank you {} for the generous donation!".format(f_name), '\n')
+    return donors
```

Comment:
No need to return donors here

Reply:
I think I need to return donors because this function updates and adds new donors. If not returned, the main() will not have the latest donor list.

Reply:
in this particular assignment the main data structure `donors` is defined at the module level, at the top of the file.

that way it is a global data structure - it simplifies the code a lot.